### PR TITLE
document transaction commands and concept

### LINKED ISSUE #1022: Document `START TRANSACTION/BEGIN`, `COMMIT` commands
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/10735

### Which part(s) of the docs might be affected or should be updated? And how?

Document the experimental `START TRANSACTION/BEGIN`, `COMMIT` commands. `START TRANSACTION` and `BEGIN` have the same functionality. 

### Reference

https://www.postgresql.org/docs/current/sql-start-transaction.html
https://www.postgresql.org/docs/current/sql-begin.html
https://www.postgresql.org/docs/current/sql-commit.html

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

 Add a brief summary of read-only transactions in RisingWave, and describe the commands (`BEGIN` `COMMIT` `START TRANSACTION`) in a bare-minimum way. We can enrich the content as the feature becomes more mature.

 Update: Add `READ ONLY` to the command pages, and updated the list of statements not allowed within a transaction.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #1022

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/commands/sql-start-transaction.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+:::caution Experimental feature
+
+Read-only transactions is currently an experimental feature in RisingWave, and its functionality is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
+
+:::
+
+## Syntax
+
+```sql
+BEGIN;
```
Comment (fuyufjh) [inline docs/sql/commands/sql-start-transaction.md]: Shall we document the read-only transaction as well? cc. @BugenZhao 

```
start transaction read only;
```

related https://github.com/risingwavelabs/risingwave/pull/10902

File: docs/concepts/transactions.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+id: transactions
+title: Transactions
+slug: /transactions
+---
+
+Transactions in databases refer to logical units of work that consist of one or more database operations. A transaction is a sequence of database operations, such as reads (queries) and writes (updates or inserts), that are treated as a single indivisible and consistent unit. The main purpose of transactions is to ensure data integrity and maintain the ACID (Atomicity, Consistency, Isolation, Durability) properties of the database.
+
+RisingWave supports read-only transactions, where all reads within a transaction are executed against the consistent Hummock snapshot. Hummock, the LSM-Tree-based storage engine in RisingWave, is specifically optimized for streaming workloads.
```
Comment (fuyufjh) [inline docs/concepts/transactions.md]: This is not accurate. 

RisingWave supports both read and write transaction but not fully ACID. 

- For writes, `Atomicity` is guaranteed, i.e. Data written in one transaction take effect atomically (as long as it's not too large) This work is done by @chenzl25 https://github.com/risingwavelabs/risingwave/pull/10077
- For reads, as mentioned in the docs, all reads are consistent. This is an intrinsic feature from the very beginning, but the support for multiple statements (i.e. `BEGIN...COMMIT`) is formalized by @BugenZhao  https://github.com/risingwavelabs/risingwave/pull/10902

File: docs/sql/commands/sql-commit.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+id: sql-commit
+title: COMMIT
+description: Commit the current transaction.
+slug: /sql-commit
+---
+
+RisingWave supports read-only transactions. You can use the `COMMIT` command to commit the current transaction. For more information about transactions in RisingWave, see [Transactions](/concepts/tranactions.md).
```
Comment (fuyufjh) [inline docs/sql/commands/sql-commit.md:8]: Ditto.

File: docs/concepts/transactions.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+:::caution Experimental feature
+
+Read-only transactions is currently an experimental feature, and its functionality is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
+
+:::
+
+To initiate a transaction, use either the `START TRANSACTION` or `BEGIN` command. Subsequently, you can execute queries to read data from the consistent snapshot. To finalize the transaction and submit the queries as a single unit, use the `COMMIT` command.
+
+Please note that read-only transactions do not permit write operations such as insertions, updates, or deletions. That means that data modifications are not allowed while a transaction is initiated but not yet committed.
```
Comment (BugenZhao) [inline docs/concepts/transactions.md]: Also, all DDLs and `USER` or `GRANT` statements are not allowed.

File: docs/sql/commands/sql-start-transaction.md (hunk 1 above)
Comment (BugenZhao) [reply docs/sql/commands/sql-start-transaction.md]: Yes. It's worth noting that `BEGIN` or `START TRANSACTION` is by default in read-write mode for compatibility, and no transaction will be actually started. We should instruct users to write `READ ONLY` explicitly.

```
dev=> start transaction;
NOTICE:  Read-write transaction is not supported yet. Please specify `READ ONLY` to start a read-only transaction.
For compatibility, this statement will still succeed but no transaction is actually started.
START_TRANSACTION
dev=> start transaction read only;
START_TRANSACTION
dev=*>
```

File: docs/concepts/transactions.md (hunk 2 above)
Comment (hengm3467) [reply docs/concepts/transactions.md]: @fuyufjh @BugenZhao Shall we only disclose read-only transactions now, as read-write transactions are partially done (correct me if I'm wrong)? 

This is based on this tracking issue: https://github.com/risingwavelabs/risingwave/issues/10736.

Comment (BugenZhao) [reply docs/concepts/transactions.md]: > read-write transactions are partially done

I guess this feature is not planned and will not be implemented in the near future.

**Comment by hengm3467** [comment]
@BugenZhao I updated the topics based on your comments. Please take a look again. Thanks!

File: docs/sql/commands/sql-start-transaction.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+id: sql-start-transaction
+title: START TRANSACTION
+description: Start a transaction.
+slug: /sql-start-transaction
+---
+
+RisingWave supports read-only transactions. You can use the `START TRANSACTION READ ONLY` command to start a read-only transaction. For more information about transactions in RisingWave, see [Transactions](/concepts/tranactions.md).
```
Comment (BugenZhao) [inline docs/sql/commands/sql-start-transaction.md:8]: What about noting here that `START TRANSACTION` is identical to `BEGIN`?

File: docs/sql/commands/sql-start-transaction.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+:::caution Experimental feature
+
+Read-only transactions is currently an experimental feature in RisingWave, and its functionality is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
+
+:::
+
+## Syntax
+
+```sql
+BEGIN;
```
Comment (BugenZhao) [inline docs/sql/commands/sql-start-transaction.md]:
```suggestion
START TRANSACTION;
```

File: docs/concepts/transactions.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+:::
+
+To initiate a transaction, use either the `START TRANSACTION READ ONLY` or `BEGIN READ ONLY` command. Subsequently, you can execute queries to read data from the consistent snapshot. To finalize the transaction and submit the queries as a single unit, use the `COMMIT` command.
+
+Please note that data modifications are not allowed while a transaction is initiated but not yet committed. The statements listed below are not allowed within a transaction:
+
+- All DDL statements (`CREATE`, `ALTER`, and `DROP`)
+- Most of DML statements (`INSERT`, `UPDATE`, and `DELETE`)
+- Statements related to `USER`. This category may overlap with DDL statements.
+- `GRANT`
```
Comment (BugenZhao) [inline docs/concepts/transactions.md]: All privilege-related statements, including `GRANT` and `REVOKE`.

**Comment by hengm3467** [comment]
@BugenZhao All comments were implemented. Thank you so much!

File: docs/sql/commands/sql-commit.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+:::caution Experimental feature
+
+Read-only transactions is currently an experimental feature in RisingWave, and its functionality is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
+
+:::
+
+## Syntax
+
+```sql
+BEGIN;
```
Comment (emile-00) [inline docs/sql/commands/sql-commit.md]: Syntax shows `BEGIN` instead of `COMMIT`

File: docs/sql/commands/sql-start-transaction.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
+---
+id: sql-start-transaction
+title: START TRANSACTION
+description: Start a transaction.
+slug: /sql-start-transaction
+---
+
+RisingWave supports read-only transactions. You can use the `START TRANSACTION READ ONLY` command to start a read-only transaction. For more information about transactions in RisingWave, see [Transactions](/concepts/tranactions.md).
+
+The `STRAT TRANSACTION` command starts the read-write transaction mode, which is not supported yet in RisingWave. For compatibility reasons, this command will still succeed but no transaction is actually started. That is why you need to specify the `READ ONLY` option to start a transaction in read-only mode.
```
Comment (emile-00) [inline docs/sql/commands/sql-start-transaction.md]: Typo: `STRAT` -> `START`

**Review by emile-00 — APPROVED** [review]

**Pull request merged**